Doc (#245)
megaease/easeagent: doc/spring-petclinic-demo.md

# Spring Petclinic Demo

You need Java 1.8+ and easeagent:

> If you don't already have easeagent, get it and set $EASE_AGENT_PATH: [EaseAgent](../README.md#get-and-set-environment-variable)

Reviewer: ```suggestion
If you haven't `easeagent`, get it and set $EASE_AGENT_PATH: [EaseAgent](../README.md#get-and-set-environment-variable)
```

### Step 1
Building the demo application.
```
$ git clone https://github.com/akwei/spring-petclinic-microservices.git
$ cd spring-petclinic-microservices
$ mvn -DskipTests=true package
```

> 
> export EASE_AGENT_PATH=/Users/beyond/IdeaProjects/easeagent_dir/easeagent/build/target

Reviewer: ```suggestion
Setup the `EASE_AGENT_PATH` environment variable
```shell
 $ export EASE_AGENT_PATH=/Users/beyond/IdeaProjects/easeagent_dir/easeagent/build/target
```
```

Author: > 

Sorry, this is a line I added by mistake when testing

### Step 2
Run the demo application with EaseAgent.
```

> # Open another console

Reviewer: ```suggestion
# Run Spring Cloud Config Server
```

$ java -jar spring-petclinic-config-server/target/spring-petclinic-config-server-2.4.2.jar

> # Open another console

Reviewer: ```suggestion
# Run Spring Cloud Service Discovery Server
```

$ java -jar spring-petclinic-discovery-server/target/spring-petclinic-discovery-server-2.4.2.jar

> # Open another console

Reviewer: ```suggestion
# Run Spring Cloud Application - Vets Service
```

$ export EASE_AGENT_PATH=[Replace with agent path]
$ cp $EASE_AGENT_PATH/agent.properties $EASE_AGENT_PATH/vets-agent.properties
$ # vi $EASE_AGENT_PATH/vets-agent.properties replease name to "petclinic-vets-service"
$ java "-javaagent:${EASE_AGENT_PATH}/easeagent.jar" -Deaseagent.config.path=${EASE_AGENT_PATH}/vets-agent.properties -Deaseagent.server.port=9900 -jar spring-petclinic-vets-service/target/spring-petclinic-vets-service-2.4.2.jar

> # Open another console

Reviewer: ```suggestion
# Run Spring Cloud Application - Visits Service
```

$ export EASE_AGENT_PATH=[Replace with agent path]
$ cp $EASE_AGENT_PATH/agent.properties $EASE_AGENT_PATH/visits-agent.properties
$ vi $EASE_AGENT_PATH/visits-agent.properties # replease name to "petclinic-visits-service"
$ java "-javaagent:${EASE_AGENT_PATH}/easeagent.jar" -Deaseagent.config.path=${EASE_AGENT_PATH}/visits-agent.properties -Deaseagent.server.port=9901 -jar spring-petclinic-visits-service/target/spring-petclinic-visits-service-2.4.2.jar

> # Open another console

Reviewer: ```suggestion
# Run Spring Cloud Application - Customers Service
```

$ export EASE_AGENT_PATH=/[Replace with agent path]
$ cp $EASE_AGENT_PATH/agent.properties $EASE_AGENT_PATH/customers-agent.properties
$ vi $EASE_AGENT_PATH/customers-agent.properties # replease name to "petclinic-customers-service"
$ java "-javaagent:${EASE_AGENT_PATH}/easeagent.jar" -Deaseagent.config.path=${EASE_AGENT_PATH}/customers-agent.properties -Deaseagent.server.port=9902 -jar spring-petclinic-customers-service/target/spring-petclinic-customers-service-2.4.2.jar

> # Open another console

Reviewer: ```suggestion
# Run Spring Cloud API Gateway
```
